ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh

```diff
@@ -92,6 +93,11 @@ if [ -z "${VNET_NAME}" ]; then
     echo "Did not find ${vnet_file}!"
     exit 1
   fi
+elif [[ "${VNET_NAME}" == "PRESERVED_FROM_CLUSTER" ]]; then
```

Comment:
I guess we do not have to use `PRESERVED_FROM_CLUSTER` here, just judge if `${SHARED_DIR}/metadata.json` exits or not, if it exists, we can set vnet name and subnet for bastion host. 

The priority would be like `VNET_NAME` > `metadata.json`